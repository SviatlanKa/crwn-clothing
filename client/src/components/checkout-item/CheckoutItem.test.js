import React from 'react';
import { shallow } from 'enzyme';
import { CheckoutItem } from './CheckoutItem';

describe('testing CheckoutItem Component', () => {
    let wrapper;
    const cartItem = {
        imageUrl: 'www.images/blue-hat.jpg',
        name: 'Blue Hat',
        quantity: 3,
        price: 100
    };
    let addItem, removeItem, clearItem;

    beforeEach(() => {
        addItem = jest.fn();
        removeItem = jest.fn();
        clearItem = jest.fn();
        const mockProps = { cartItem, addItem, removeItem, clearItem };
        wrapper = shallow(<CheckoutItem { ...mockProps } />);
    });

    it('render CheckoutItem Component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call removeItem when click', () => {
        wrapper.find('.left').simulate('click');
        expect(removeItem).toHaveBeenCalled();
    });
    it('should call addItem when click', () => {
        wrapper.find('.right').simulate('click');
        expect(addItem).toHaveBeenCalled();
    });
    it('should call clearItem when click', () => {
        wrapper.find('.remove-button').simulate('click');
        expect(clearItem).toHaveBeenCalled();
    });
});