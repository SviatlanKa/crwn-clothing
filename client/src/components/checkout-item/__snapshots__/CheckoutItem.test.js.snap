// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CheckoutItem Component render CheckoutItem Component 1`] = `
<div
  className="checkout-item"
>
  <div
    className="image-container"
  >
    <img
      alt="item"
      src="www.images/blue-hat.jpg"
    />
  </div>
  <span
    className="name"
  >
    Blue Hat
  </span>
  <span
    className="quantity"
  >
    <div
      className="arrow left"
      onClick={[Function]}
    >
      ❮
    </div>
    <span
      className="value"
    >
      3
    </span>
    <div
      className="arrow right"
      onClick={[Function]}
    >
      ❯
    </div>
  </span>
  <span
    className="price"
  >
    100
  </span>
  <div
    className="remove-button"
    onClick={[Function]}
  >
    ✕
  </div>
</div>
`;
